multi_physics/QED/QED_table_generator/table_generator.cpp

```diff
@@ -198,71 +298,440 @@ void generate_quantum_sync_photem_table(
 
     const auto raw_data = table.serialize();
 
-    std::ofstream of{file_name};
+    std::ofstream of{file_name + "_photem.bin"};
     of.write (raw_data.data(),raw_data.size());
     of.close();
 
     write_csv_2d_table(table, qs_params.chi_part_min*0.1f, qs_params.chi_part_max*10.f,
         std::numeric_limits<RealType>::epsilon(), RealType(1.0), qs_params.chi_part_how_many*3,
-        qs_params.frac_how_many*3, true, false, file_name + ".csv");
+        qs_params.frac_how_many*3, true, false, file_name + "_photem.csv");
+}
+
+/**
+* Parses arguments to set parameters for Breit-Wheeler lookup tables generation
+*
+* @tparam RealType the floating point type to be used
+* @param[in] args the command line arguments
+* @return a struct containing the parameters to generate Breit-Wheeler lookup tables
+*/
+template<typename RealType>
+BreitWheelerTableParams<RealType>
+parse_breit_wheeler_params(std::map<std::string, std::string> args)
+{
+    BreitWheelerTableParams<RealType> params;
+
+    params.chi_min = px_bw::default_chi_phot_min<RealType>;
+    params.chi_max = px_bw::default_chi_phot_max<RealType>;
+    params.chi_size = px_bw::default_chi_phot_how_many;
+    params.frac_size = px_bw::default_frac_how_many;
+
+    auto s_cmin = args.find(CMD_CHI_MIN);
+    if(s_cmin != args.end())
+        params.chi_min = static_cast<RealType>(std::stod(s_cmin->second));
```

Comment:
It looks like `stod` does not fail for inputs like "0.6AAA" (it will return 0.6 in this case), which is not 100% clean. I don't think that's a big issue so I'm fine with leaving it as it is. If you wanted to change this though, you could use the second argument of `stod` and check that it's equal to the size of the input string.

Reply:
You are right.
I've added these two wrappers to replace direct calls to `stod` and `stoi`:
```
/**
* Wraps the call to std::stod, checking if the string has been correctly parsed.
* If conversion fails, it prints an error message and exits the program.
*
* @param[in] str the string to be parsed
* @return std::stod(str) if conversion succeeds
*/
double stod_wrapper(const std::string& str)
{
    size_t idx = 0;
    double val = std::stod(str, &idx);
    if (idx != str.length()){
        print_error("Failed to parse '" + str + "' as a floating point number!");
        exit(EXIT_FAILURE);
    }
    return val;
}

/**
* Wraps the call to std::stoi, checking if the string has been correctly parsed.
* If conversion fails, it prints an error message and exits the program.
*
* @param[in] str the string to be parsed
* @return std::stoi(str) if conversion succeeds
*/
int stoi_wrapper(const std::string& str)
{
    size_t idx = 0;
    int val = std::stoi(str, &idx);
    if (idx != str.length()){
        print_error("Failed to parse '" + str + "' as an integer number!");
        exit(EXIT_FAILURE);
    }
    return val;
}

```